memgpt/server/server.py

```diff
@@ -600,20 +600,52 @@ def get_agent_messages(self, user_id: uuid.UUID, agent_id: uuid.UUID, start: int
         if start < 0 or count < 0:
             raise ValueError("Start and count values should be non-negative")
 
-        # Reverse the list to make it in reverse chronological order
-        reversed_messages = memgpt_agent.messages[::-1]
+        if start + count < len(memgpt_agent.messages):  # messages can be returned from whats in memory
+            # Reverse the list to make it in reverse chronological order
+            reversed_messages = memgpt_agent.messages[::-1]
+            # Check if start is within the range of the list
+            if start >= len(reversed_messages):
+                raise IndexError("Start index is out of range")
+
+            # Calculate the end index, ensuring it does not exceed the list length
+            end_index = min(start + count, len(reversed_messages))
+
+            # Slice the list for pagination
+            paginated_messages = reversed_messages[start:end_index]
+
+            # convert to message objects:
+            messages = [memgpt_agent.persistence_manager.json_to_message(m) for m in paginated_messages]
+        else:
```

Comment:
@sarahwooders did we want to add a field indicating in-context / out-of-context here?